Add the BUILD_PROGRAMS option to disable programs build

By default, PolarSSL builds and installs a large set of companions
programs, which in some cases are not useful. This patch adds the
BUILD_PROGRAMS option which allows to disable the build and
installation of such programs when not needed.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: polarssl-1.1.1/CMakeLists.txt
===================================================================
--- polarssl-1.1.1.orig/CMakeLists.txt
+++ polarssl-1.1.1/CMakeLists.txt
@@ -33,7 +33,11 @@
   add_subdirectory(tests)
 endif(CMAKE_COMPILER_IS_GNUCC AND BUILD_TESTS)
 
-add_subdirectory(programs)
+option(BUILD_PROGRAMS "Build programs." ON)
+
+if(BUILD_PROGRAMS)
+  add_subdirectory(programs)
+endif(BUILD_PROGRAMS)
 
 ADD_CUSTOM_TARGET(apidoc
                   COMMAND doxygen doxygen/polarssl.doxyfile
